R/generate_dist.R

```diff
@@ -336,6 +345,34 @@
   new
 }
 
+# Internal function called by .generate - returns non-random data
+#
+# @param n The number of observations required in the data set
+# @param fn String that specifies the custom function
+# @param args String of comma delimited arguments that will be passed to fn
+# @param dtSim Incomplete simulated data.table
+# @return A data.frame column  with the updated simulated data
+
+.gencustom <- function(n, fn, args, dtSim, envir) {
+  
+  # new <- .evalWith(formula, .parseDotVars(formula, envir), dtSim, n)
+  
+  args <- gsub("\\s+", "", args) # remove any spaces
+  arg_l <- as.list(unlist(strsplit(args, ",")))
```

Comment:
```suggestion
   arg_l <- args |> strsplit(",") |> unlist(recursive = FALSE) |> trimws()
```
with recursive it only strips one layer of list, that makes `as.list` superfluos, `trimws` removes leading and trailing whitespace. That means it wont remove the spaces around = if that's passed in, not sure if that works with the rest of the code, so maybe `gsub` is better :)

Reply:
I really need to strip out all spaces - and create a list of lists of two items (the variable name and the formula) - so I think I'll keep it as I wrote it.